### DISCUSSION TITLE
Set stream bitrate

### DISCUSSION THREAD

**Discussion by hypeARM** [rtl-airband/RTLSDR-Airband#340] [Ideas]
It would be interesting to be able to set a higher bitrate for the stream output

This would help to reduce the transmission delay as it would transmit with more bandwidth

**Comment by charlie-foxtrot** [comment]
The output bitrate on the UDP stream is correlated to the wave rate that in turn is related to the modulations enabled (compiling AM-only gives 8 kHz, building in NFM gives 16 kHz).

There really shouldn't be a delay internal to this program, could you elaborate where you are seeing a delay and how/why increasing bandwidth would help?

**Reply by hypeARM** [reply]
It is not an internal delay, but to reduce the delay in the online stream, icecast

Do you think increasing the bitrate would decrease the delay?

I have a stream with 320kbps with 6 seconds encode/decode, while the rtl_airband exceeds 30 seconds

**Reply by hamav8tor** [reply]
Latency is largely player-dependent, and largely depends on the buffer size in the player. Since many consume these streams via browsers, the buffer size is not controllable. Since it is a fixed size, a lower bitrate stream takes longer to fill the buffer, hence the longer delay. Icecast has a burst-on-connect feature, but while that initially fills the player buffer faster for a faster connection, it results in more latency than using a smaller burst-on-connect burst size and just incurring the pain of a slightly longer initial connection and initial play latency.

**Reply by charlie-foxtrot** [reply]
Yes, you may be hitting a buffering in icecast, but there are configs in icecast that can be adjusted to reduce / eliminate this.  I am running icecast locally with rtl-airband and have minimal delay when listening from another computer.

See `queue-size`, `burst-size` and other parameters here: https://icecast.org/docs/icecast-latest/config-file.html

**Reply by hypeARM** [reply]
So increasing transmission bitrate wouldn't do much good?

The server I send stream is Radarbox

**Reply by hamav8tor** [reply]
All other things being equal, increasing transmission rate *does* decrease latency.

**Reply by hypeARM** [reply]
So my idea would be theoretically valid, since increasing the transmission rate would progressively decrease the latency between the client and the server...

**Reply by charlie-foxtrot** [reply]
Yes, if you dont have control of the server config, increasing the MP3 data size may be your only option . . . . seems like a real waste though.

Given the audio samples going into creating the MP3 is a max of 16 kHz, I wonder if the MP3 compression will keep the file size small even if you crank up the "bitrate" 🤔

**Reply by hypeARM** [reply]
I don't have access there, and it seems to be difficult for them to change this option just for "me"

I'll talk to support and see what they say

What `queue-size`, `burst-size` numbers would be ideal?

**Reply by charlie-foxtrot** [reply]
icecast does allow these parameters to be set per-stream (aka mount).  I have `burst-size` set to `0` and `queue-size` set to the default of 512k.  If they are up for experimenting, if they can drop the `burst-size` down to `0` just for your mount that would be informative.  Like mentioned by @hamav8tor the buffering is done client-side so you may see different results on different clients.  FWIW I'm listening in Safari.

**Reply by hypeARM** [reply]
Interesting, I'll ask them to set `burst-size ` and `queue-size ` just for my mount

**Reply by charlie-foxtrot** [reply]
@hypeARM curious if they were able to adjust those parameters and if that showed an improvement?

**Reply by hypeARM** [reply]
Hi @charlie-foxtrot 

To this day they haven't answered me, it seems they don't want to do anything

Were silent...
